programs/stake/src/stake_state.rs

```diff
@@ -978,13 +978,20 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
                         // split account. Since split accounts retain the state of their source
                         // account, this prevents any magic activation of stake by prefunding the
                         // split account.
-                        (
-                            lamports.saturating_sub(meta.rent_exempt_reserve),
+                        // The new split stake also needs to ignore any positive delta between the
+                        // original rent_exempt_reserve and the split_rent_exempt_reserve, in order
+                        // to prevent magic activation of stake by splitting between accounts of
+                        // different sizes.
+                        let remaining_stake_delta =
+                            lamports.saturating_sub(meta.rent_exempt_reserve);
+                        let split_stake_amount = std::cmp::min(
                             lamports - split_rent_exempt_reserve,
-                        )
+                            remaining_stake_delta,
+                        );
```

Comment:
Hmm, I only tend to complicate things further down, but this creates a different loophole where locked stack can legally unlocked pretty easily by splitting back and forth large-sized accounts:

```
Firstly, split (or fully-split to) largest-sized account from your normal locked stake account.
Say, 1000 SOL largest-sized stake account with 72.981_780_480 rent exempt reserve (MAX_PERMITTED_DATA_LENGTH):

Then, split back into very small-sized account (200 data len; Rent Exempt Reserve: 0.00228288 SOL)
so, this new min will look like this:

std::cmp::min(
    1000 - 0.002_282_880, // lamports - split_rent_exempt_reserve
    1000 - 72.981_780_480, // remaining_stake_delta
) => // 927.01821952

Then, the rest (72.981_780_480) should be withdrawal, meaning these are now in your free hands. :)

And repeat this as you like
```

Reply:
> Hmm, I only tend to complicate things further down, but this creates a different loophole where locked stake can legally unlocked pretty easily by splitting back and forth large-sized accounts:
> 
> ```
> Firstly, split (or fully-split to) largest-sized account from your normal locked stake account.
> Say, 1000 SOL largest-sized stake account with 72.981_780_480 rent exempt reserve (MAX_PERMITTED_DATA_LENGTH):
> 
> Then, split back into very small-sized account (200 data len; Rent Exempt Reserve: 0.00228288 SOL)
> so, this new min will look like this:
> 
> std::cmp::min(
>     1000 - 0.002_282_880, // lamports - split_rent_exempt_reserve
>     1000 - 72.981_780_480, // remaining_stake_delta
> ) => // 927.01821952
> 
> Then, the rest (72.981_780_480) should be available for withdrawal, meaning these are now in your free hands. :)
> 
> And repeat this as you like. One iteration should cost  only 1 or 2 sig fees.
> ```

oop. I was wrong. well, excess locked-up funds aren't withdrawal..

Reply:
Yep, lockups should apply regardless of delegation status 😅 Good to have that confirmed though, thanks!

Reply:
I think we _do_ have an adjacent issue here though.  Since split can pull the split account's rent_exempt_reserve out of an active stake, that leaves an avenue to magic deactivation of stake and throwing off the cluster distribution.

I think we're going to need to require the split account be pre-funded with the rent_expempt_reserve, which might be tricky ABI-wise

Reply:
Yes, that's true, but it seems less risky to me.
That said... would it be totally crazy to require that newly initialized stake accounts be the right size?

Reply:
I considered clamping the data size, but it only makes the hole smaller.  You can still get a bunch of split/merge ratchets going and deactivate an account pretty quickly